src/Commands/Link.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Stancl\Tenancy\Commands;
+
+use Illuminate\Console\Command;
+use Stancl\Tenancy\Concerns\HasATenantsOption;
+use Stancl\Tenancy\Contracts\Tenant;
+
+class Link extends Command
+{
+    use HasATenantsOption;
+
+    /**
+     * The console command signature.
+     *
+     * @var string
+     */
+    protected $signature = 'tenants:link
+                {--tenants=* : The tenant(s) to run the command for. Default: all}
+                {--relative : Create the symbolic link using relative paths}
+                {--force : Recreate existing symbolic links}
+                {--remove : Remove symbolic links}';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Create symbolic links for tenants.';
+
+    /**
+     * Execute the console command.
+     *
+     * @return void
+     */
+    public function handle()
```

Comment:
This method, as well as `links()` is pretty hard to read. @lukinovec do you think you could refactor it, perhaps decrease the complexity of the nesting, extract things into methods, add variables or comments? You don't have to do exactly those things of course, but something along those lines to improve readability. The code is very low level and crammed, so it's very hard to get an idea of what's happening here without actually reading the class for several minutes.

Reply:
The `tenant:link` command does very different things depending on the `--remove` option. If it's passed, instead of creating the symlinks, the existing ones are removed.
```php
public function handle()
{
    if ($this->option('remove')) {
        $this->removeLinks();

        return;
    }

    $this->createLinks();
}
```
Sounds like these two actions should have separate commands.

Reply:
```php
$tenants = $this->option('remove') && filled($this->option('tenants'))
            ? collect($this->option('tenants'))
            : $this->getTenants()->map(function (Tenant $tenant) {
                return $tenant->getTenantKey();
            });
```

Using the tenants from the option only when removing the links doesn't make much sense to me. If we specify `tenants` in the option, the command should work with these. Something like this should be sufficient:
```php
$tenants = $this->option('tenants') ?? $this->getTenants()->map->getTenantKey()
```

Reply:
I refactored the command https://github.com/archtechx/tenancy/pull/909/commits/a45f02445cfd4e21268ec68b5a8c6b075aa8be1f